packages/@o3r/core/schematics/convert-component/index.ts

```diff
@@ -0,0 +1,63 @@
+import { chain, noop, Rule, Tree } from '@angular-devkit/schematics';
+import { applyEsLintFix, isNgClassComponent, isO3rClassComponent } from '@o3r/schematics';
+import { insertImport } from '@schematics/angular/utility/ast-utils';
+import { applyToUpdateRecorder, InsertChange } from '@schematics/angular/utility/change';
+import * as ts from 'typescript';
+import { ConvertToO3rComponentSchematicsSchema } from './schema';
+
+
+/**
+ * Convert an Angular component into an Otter component
+ *
+ * @param options
+ */
+export function convertToO3rComponent(options: ConvertToO3rComponentSchematicsSchema): Rule {
+
+  const updateComponentFile: Rule = (tree: Tree) => {
+    let sourceFile = ts.createSourceFile(
+      options.path,
+      tree.readText(options.path),
+      ts.ScriptTarget.ES2020,
+      true
+    );
+
+    const recorder = tree.beginUpdate(options.path);
+    const changes = [];
+
+    changes.push(insertImport(sourceFile, options.path, 'O3rComponent', '@o3r/core'));
+
+    sourceFile = ts.createSourceFile(
```

Comment:
why do you need to reinitialize it ? the `sourceFile` is not updated when you do the `insertImport` ?